numpyro/distributions/constraints.py

```diff
@@ -282,9 +263,8 @@ class :class:`StickBreakingTransform` to transform :math:`X_i` into a
 
     def __call__(self, x):
         # we interchange step 1 and step 2.a for a better performance
-        eps = np.finfo(x.dtype).eps
-        t = np.clip(np.tanh(x), a_min=(-1 + eps), a_max=(1 - eps))
-        return _signed_stick_breaking_tril(t)
+        t = np.tanh(x)
+        return signed_stick_breaking_tril(t)
```

Comment:
I also move the `clip` logic to `signed_stick_breaking_tril` function because we also need to clip in the sampler.